packages/plugin/src/theia.d.ts

```diff
@@ -5377,6 +5377,17 @@ export module '@theia/plugin' {
          */
         password: boolean;
 
+        /**
+         * Selection range in the input value. Defined as tuple of two number where the
+         * first is the inclusive start index and the second the exclusive end index. When `undefined` the whole
+         * pre-filled value will be selected, when empty (start equals end) only the cursor will be set,
+         * otherwise the defined range will be selected.
+         *
+         * This property does not get updated when the user types or makes a selection,
+         * but it can be updated by the extension.
+         */
+        valueSelection: readonly [number, number] | undefined;
```

Comment:
Looking at the VS Code API `valueSelection` shouldn't be undefined. I think the solution is to ensure the field is always defined, rather than allowing it to be undefined.

Reply:
Note that there is a field named `valueSelection` that can be undefined, but it's from `InputBoxOptions` and not `InputBox` as is the case here.

Reply:
but if you look into the vscode.d.ts in the vs code source code there the field is defined as `valueSelection: readonly [number, number] | undefined;` for `InputBox` and `valueSelection?: [number, number]; for `InputBoxOptions`
`

Reply:
so this is exactly how its implementd in vscode

Reply:
You are right, I just double checked in `vscode.d.ts` and it seems to be allowed... I was solely looking at the web documentation here: https://code.visualstudio.com/api/references/vscode-api.

It seems to be an issue with their web doc, your changes are good.